### PR TITLE
Add source type get and list endpoint

### PR DESCRIPTION
**Pull request by lpichler**
WIP reason:
- [x] https://github.com/RedHatInsights/sources-api-go/pull/8

This add get and list action for source types.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ func main() {
 	// setting up the DAO functions
 	getSourceDao = getSourceDaoWithTenant
 	getApplicationTypeDao = getApplicationTypeWithoutTenant
+	getSourceTypeDao = getSourceTypeWithoutTenant
```
Comment (lindgrenj6) [inline main.go]: This is a mistake on my part, can we rename these functions like `getSourceTypeDAOWithoutTenant` for both the SourceType and ApplicationType DAO?

File: model/source_type_http.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package model
+
+import (
```
Comment (lindgrenj6) [inline model/source_type_http.go:3]: to fix these 2 errors you can run `make gci` just FYI

File: routes.go [hunk 3]
```diff
@@ -44,6 +44,10 @@ func setupRoutes(e *echo.Echo) {
 	// ApplicationTypes
 	v3.GET("/application_types", ApplicationTypeList, middleware.ParseFilter, middleware.ParsePagination)
 	v3.GET("/application_types/:id", ApplicationTypeGet)
+
+	// SourceTypes
+	v3.GET("/source_types", SourceTypeList, middleware.ParseFilter, middleware.ParsePagination)
+	v3.GET("/source_types/:id",SourceTypeGet)
```
Comment (lindgrenj6) [inline routes.go]: Could use a space after the comma here, but I think if you run `go fmt -w .` to fix it (and any other mis-formattings).

File: source_type_handlers.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+var getSourceTypeDao func(c echo.Context) (dao.SourceTypeDao, error)
+
+func getSourceTypeWithoutTenant(_ echo.Context) (dao.SourceTypeDao, error) {
+	// we do not need tenancy for source type.
+
+	return &dao.SourceTypeDaoImpl{}, nil
+}
+
+func SourceTypeList(c echo.Context) error {
+	SourceTypeDB, err := getSourceTypeDao(c)
```
Comment (lindgrenj6) [inline source_type_handlers.go]: Can we make these lowercase throughout this file? Uppercase tends to indicate that it is exported from the package.

File: source_type_handlers.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		return err
+	}
+
+	limit, offset, err := getLimitAndOffset(c)
+	if err != nil {
+		return err
+	}
+
+	sourceTypes, count, err := SourceTypeDB.List(limit, offset, filters)
+
```
Comment (lindgrenj6) [inline source_type_handlers.go]: This space can go away.

**Review by lindgrenj6 — COMMENTED** [review]
Just a few changes - some of which will probably be taken care of by `make gci` and `gofmt -w .` 

But other than that looks great, even the tests!!! WOO!

**Comment by lpichler** [comment]
@lindgrenj6 thanks for guidance and review! I addressed your feedback 👍

**Review by lindgrenj6 — APPROVED** [review]

**Pull request merged**